crates/ra_parser/src/grammar/expressions.rs

```diff
@@ -290,6 +290,25 @@ fn expr_bp(p: &mut Parser, r: Restrictions, bp: u8) -> (Option<CompletedMarker>,
         let m = lhs.precede(p);
         p.bump(op);
 
+        if is_range {
+            // test postfix_range
+            // fn foo() {
+            //     let x = 1..;
+            //     match 1.. { _ => () };
+            //     match a.b()..S { _ => () };
+            // }
+            let has_trailing_expression =
+                p.at_ts(EXPR_FIRST) && !(r.forbid_structs && p.at(T!['{']));
+            if !has_trailing_expression {
+                if op == T![..=] {
+                    p.error("expected expression to end inclusive range");
```

Comment:
another drive-by fix, `a..=` is illegal

Reply:
It is semantically illegal, but I think it's better to parse it just like `a..`, and instead handle the error during validation, similarly to how we handle [visibility on trait items](https://github.com/rust-analyzer/rust-analyzer/blob/31d01efb069065449e53cfd7d0935c4b9fecf1e3/crates/ra_syntax/src/validation.rs#L211-L229)